crates/biome_module_graph/src/module_graph.rs

```diff
@@ -27,6 +27,10 @@ pub const SUPPORTED_EXTENSIONS: &[&str] = &[
     "js", "jsx", "mjs", "cjs", "ts", "tsx", "mts", "cts", "json", "node",
 ];
 
+pub const SUPPORTED_TYPE_EXTENSIONS: &[&str] = &[
+    "d.ts", "d.mts", "d.cts", "js", "jsx", "mjs", "cjs", "ts", "tsx", "mts", "cts", "json",
```

Comment:
Why is `json` there?

Reply:
Because it's also in the regular `SUPPORTED_EXTENSIONS`, and I figured who knows, maybe we could derive a literal type from such files one day :)